pkg/controllers/node/ipam.go

```diff
@@ -152,23 +164,65 @@ func (c *ipamController) acceptScheduleRequests(stopCh <-chan struct{}) {
 		period = c.config.LeakGracePeriod.Duration / 2
 	}
 	t := time.NewTicker(period)
+	log.Info("Will run periodic IPAM sync every %s", period)
 	for {
 		// Wait until something wakes us up, or we are stopped
 		select {
-		case kvp := <-c.blockUpdate:
+		case kvp := <-c.nodeUpdate:
+			log.Debug("Node update received")
 			if kvp.Value != nil {
-				c.onBlockUpdated(kvp)
+				n := kvp.Value.(*apiv3.Node)
+				kn, err := getK8sNodeName(*n)
+				if err != nil {
+					log.WithError(err).Info("Unable to get corresponding k8s node name, skipping")
+				} else if kn != "" {
+					if current, ok := c.kubernetesNodesByCalicoName[n.Name]; !ok {
+						logrus.Debugf("Add mapping calico node -> k8s node. %s -> %s", n.Name, kn)
+						c.kubernetesNodesByCalicoName[n.Name] = kn
+					} else if current != kn {
+						logrus.Warnf("Update mapping calico node -> k8s node. %s -> %s (previously %s)", n.Name, kn, current)
+						c.kubernetesNodesByCalicoName[n.Name] = kn
+					}
+					// No change.
+				}
 			} else {
-				c.onBlockDeleted(kvp.Key.(model.BlockKey))
+				cnode := kvp.Key.(model.ResourceKey).Name
+				if _, ok := c.kubernetesNodesByCalicoName[cnode]; ok {
+					logrus.Debugf("Remove mapping for calico node %s", cnode)
+					delete(c.kubernetesNodesByCalicoName, cnode)
+				}
 			}
+			log.Debug("Node update complete")
+		case kvp := <-c.blockUpdate:
+			c.handleBlockUpdate(kvp)
+
+			// It's possible we get a rapid series of block updates in a row. Use
+			// a consolidation loop to handle "batches" of block updates before triggering a sync.
+			var i int
+			for i = 1; i < 1000; i++ {
+				select {
+				case kvp = <-c.blockUpdate:
```

Comment:
If a burst is possible you might want blockUpdate to be a buffered channel.  Otherwise the caller probably won't have chance to queue the next update before we loop and hit the default case.